### PR TITLE
chore: removal of the server folder and it's references

### PR DESCRIPTION
**Pull request by Gui-greg**

### CODE REVIEW

File: docs/getting-started/why-superlinked.md [hunk 1]
```diff
@@ -237,8 +238,7 @@ Broad queries eg “popular pants” can’t be handled by re-ranking at all, du
 ### Okay, seems like Superlinked proposes a nice approach, but
 
 1. How can I build with it at scale?
-    
-    [Superlinked Server](../run-in-production/overview.md) offers a streamlined deployment solution for Superlinked projects. With a single script, you can create a Superlinked application featuring REST endpoints and Vector Database integration. This enables quick implementation of advanced search capabilities, allowing you to focus on leveraging Superlinked's features rather than managing infrastructure, from prototype to production.
+    The [Superlinked Server](../run-in-production/overview.md) is a deployable component available as a [Python package on PyPI](https://pypi.org/project/superlinked-server/), designed to enhance the operation of Superlinked by providing a RESTful API for management. This package streamlines the integration of Superlinked's sophisticated search functionalities into existing applications by offering REST endpoints and Vector Database connectivity. It enables developers to focus on leveraging Superlinked's capabilities without the burden of infrastructure management, from initial prototype to full-scale production.
```
Comment (lauralehoczki11) [inline docs/getting-started/why-superlinked.md]:
```suggestion
    The [Superlinked Server](../run-in-production/overview.md) is a deployable component available as a [Python package on PyPI](https://pypi.org/project/superlinked-server/), designed to enhance the operation of Superlinked by providing a RESTful API for communicating with your application. This package streamlines the integration of Superlinked's sophisticated search functionalities into existing applications by offering REST endpoints and Vector Database connectivity. It enables developers to focus on leveraging Superlinked's capabilities without the burden of infrastructure management, from initial prototype to full-scale production.
```

the "API for management" is eerily similar to MAPI

File: README.md [hunk 2]
```diff
@@ -174,9 +180,9 @@ result.to_pandas()
...
 
-Run a single command to deploy your Superlinked project as a REST API Server locally or in your cloud with [Superlinked Server](https://github.com/superlinked/superlinked/tree/main/server). Get data ingestion and query APIs, embedding model inference and deep vector database integrations for free!
+With a single command you can run Superlinked as a REST API Server locally or in your cloud with [Superlinked Server](https://pypi.org/project/superlinked-server). Get data ingestion and query APIs, embedding model inference and deep vector database integrations for free!
 
-Unify your evaluation, ingestion and serving stacks with a single declarative python codebase. Superlinked enables this by letting you define your data schema, vector indexes and the compute DAG that links them all at once and then chose the right executor for the task - in-memory, multi-worker server or batch.
+Unify your evaluation, ingestion and serving stacks with a single declarative python codebase. Superlinked enables this by letting you define your data schema, vector indexes and the compute DAG that links them all at once and then chose the right executor for the task - in-memory, server or batch.
```
Comment (lauralehoczki11) [inline README.md]:
```suggestion
Unify your evaluation, ingestion and serving stacks with a single declarative python codebase. Superlinked enables this by letting you define your data schema, vector indexes and the compute DAG that links them all at once and then chose the right executor for the task - in-memory or server.
```

- not really, batch is proprietary

**Review by lauralehoczki11 — APPROVED** [review]

File: docs/run-in-production/setup/configuring-your-app.md [hunk 3]
```diff
@@ -89,8 +89,6 @@ In this document, you set up your source, which acts as the entry point for your
...
-
 You can find more detailed information and examples of various features in the [Superlinked feature notebooks](https://github.com/superlinked/superlinked/tree/main/notebook/feature). The [basic_building_blocks.ipynb](https://github.com/superlinked/superlinked/blob/main/notebook/feature/basic_building_blocks.ipynb) notebook provides a comprehensive guide on the basic structure and how to use it, while the other notebooks cover various features of the Superlinked library.
 
 In this deployment setup, you are not required to define any computations as you would in the [basic_building_blocks.ipynb](https://github.com/superlinked/superlinked/blob/main/notebook/feature/basic_building_blocks.ipynb) notebook. Instead, your focus will be on defining the schema, the text similarity space, the index, the query, the REST source, and the executor.
```
Comment (AndreyPikunov) [inline docs/run-in-production/setup/configuring-your-app.md:94]: Comment for line:214 which is out of commentable area. The line says:
> GPU acceleration is available exclusively for text embedding processes and depends on explicit activation.

later line:224 says:

> GPU acceleration is most effective for large batches of text or image embeddings.

don't we support GPU acceleration for OpenCLIP models used for image embeddings in our framework?

File: server/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Superlinked Server
+
+We've updated the release structure of the Superlinked Server. It is now available as a Python library hosted on the public PyPI repository. For installation and more details, visit the [Superlinked Server PyPI page](https://pypi.org/project/superlinked-server/).
```
Comment (AndreyPikunov) [inline server/README.md]: release structure -> release strategy?

**Review by AndreyPikunov — APPROVED** [review]
One question and one wording suggestion.

**Review by teddego — APPROVED** [review]

File: docs/run-in-production/setup/configuring-your-app.md (hunk 3 above)
Comment (Gui-greg) [reply docs/run-in-production/setup/configuring-your-app.md:94]: You are right! At the time of writing we had no ImageSpace! Fixing it, thank you!

**Pull request merged**